### PR TITLE
Added chat buffer, now memorizes chat

### PR DESCRIPTION
**Pull request by AssassinTee**
Chat is memorized, until you send it. If you switch to another chat-mode, the buffer will be cleared

### CODE REVIEW

**Comment by Dune-jr** [comment]
It works but there is no way of deleting the whole line other than by pressing Escape, so not being able to do that without changing chat modes is a bit counter-inuititve and clunky. Maybe Escape should keep its behavior and mouse clicks should allow a "soft exit"? Also I don't think messages should survive changing servers.

**Comment by AssassinTee** [comment]
- Delete buffer when leaving server
- Use mouse1 instead of Escape for the saving buffer
wip

**Comment by AssassinTee** [comment]
You can now come out of the chat with mouse1 and mouse2 (hook and fire) and your text is buffered and not lost.
The buffer is cleared when disconnecting.

File: src/game/client/components/chat.cpp [hunk 1]
```diff
@@ -410,6 +417,17 @@ bool CChat::OnInput(IInput::CEvent Event)
...
+	//Handle Chat Buffer
+	if(Event.m_Flags&IInput::FLAG_PRESS && (Event.m_Key == KEY_RETURN || Event.m_Key == KEY_KP_ENTER))
+	{
+		ClearChatBuffer();
+	}
+	else if(Event.m_Key != KEY_MOUSE_1 && Event.m_Key != KEY_MOUSE_2)
+	{
+		//Save Chat Buffer
+		m_ChatBufferMode = m_Mode;
+		str_copy(m_ChatBuffer, m_Input.GetString(), m_Input.GetLength()+1);
```
Comment (oy) [inline src/game/client/components/chat.cpp]: 3rd parameter should be size of m_ChatBuffer

Comment (AssassinTee) [reply src/game/client/components/chat.cpp]: okay

**Pull request merged**